### PR TITLE
Extend PPA Exposure Risk Metadata with PT (EXPOSUREAPP-6722)

### PR DESCRIPTION
**Pull request by kolyaopahle**
### Description
This PR adds the new information gathered by presence tracing to the ppa collection set,
all values that were previously collected for BLE exposures are now also collected independently for PT exposures and logged to the backend,

i did quite a few renamings at the old values but kept the underlying db and prop id's the same to allow for backwards compatibility.

### Steps to reproduce
1. Produce some analytics data the same way as before the changes
2. Check (using dev tools) if analytics data also contains PT fields

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionCollector.kt [hunk 1]
```diff
@@ -4,9 +4,11 @@ import de.rki.coronawarnapp.datadonation.analytics.common.calculateDaysSinceMost
 import de.rki.coronawarnapp.datadonation.analytics.common.toMetadataRiskLevel
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionCollector.kt:4]: Please do not change anything about key submission. I'm doing this in a different PR

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionDonor.kt [hunk 2]
```diff
@@ -40,10 +40,16 @@ class AnalyticsKeySubmissionDonor @Inject constructor(
         PpaData.PPAKeySubmissionMetadata.newBuilder()
             .setAdvancedConsentGiven(repository.advancedConsentGiven)
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionDonor.kt:41]: same here

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/registeredtest/TestResultDataCollector.kt [hunk 3]
```diff
@@ -3,6 +3,7 @@ package de.rki.coronawarnapp.datadonation.analytics.modules.registeredtest
 import de.rki.coronawarnapp.coronatest.server.CoronaTestResult
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/registeredtest/TestResultDataCollector.kt:3]: Please do not change anything about test result

**Review by chiljamgossow — CHANGES_REQUESTED** [review]
Please limit your chages to the Exposure Risk Metadata only

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelResultExtensions.kt [hunk 4]
```diff
@@ -17,6 +18,16 @@ fun List<EwRiskLevelResult>.tryLatestEwResultsWithDefaults(): DisplayableEwRiskR
     )
 }
 
+fun List<PtRiskLevelResult>.getLastCalculatedWithDefaults(): PtRiskLevelResult {
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelResultExtensions.kt:21]: What do you need this for? Defaults are provided in the RiskLevelStorage

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/storage/RiskLevelStorage.kt [hunk 5]
```diff
@@ -69,6 +70,14 @@ interface RiskLevelStorage {
      */
     val ptDayRiskStates: Flow<List<PresenceTracingDayRisk>>
 
+    /** PRESENCE TRACING RISK RESULT
+     * The newest 2 results.
+     * Used only for analytics
+     * Can be 0-2 entries.
+     * Newest item first.
+     */
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/storage/RiskLevelStorage.kt:78]: Please use latestAndLastSuccessfulCombinedEwPtRiskLevelResult instead. It has the last successful pt and ew calculation with day risks which are needed to retrieve some of the data you are using

Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/storage/RiskLevelStorage.kt:78]: val lastResult = riskLevelStorage
            .latestAndLastSuccessfulCombinedEwPtRiskLevelResult
            .first()
            .lastSuccessfullyCalculated
            .ewRiskLevelResult

Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/storage/RiskLevelStorage.kt:78]: to access ew and pt RiskLevelResult i would need to make them public

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/PtRiskLevelResult.kt [hunk 6]
```diff
@@ -50,4 +50,11 @@ data class PtRiskLevelResult(
...
         checkInWarningOverlaps?.size ?: 0
     }
+
+    val lastRiskEncounterAt: LocalDate?
+        get() = if (riskState == RiskState.INCREASED_RISK) {
+            mostRecentDateWithHighRisk
+        } else {
+            mostRecentDateWithLowRisk
+        }
 }
```
[the remark this answers is not in the document]
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/PtRiskLevelResult.kt:60]: done :)

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelResultExtensions.kt (hunk 4 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelResultExtensions.kt:21]: took it from the ew implementation but is now removed

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/storage/RiskLevelStorage.kt (hunk 5 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/storage/RiskLevelStorage.kt]: did this, if anyone complains we can discuss it

**Review by BMItr — APPROVED** [review]
lgtm

**Review by chiljamgossow — APPROVED** [review]
LGTM

**Pull request merged**